### PR TITLE
Multi-tier masternode functionality fixed

### PR DESCRIPTION
**Pull request by kemi04**
Multi-tier masternode rewards now should reflect the tier ratios between masternodes.

### CODE REVIEW

File: src/masternodeman.cpp [hunk 1]
```diff
@@ -494,10 +493,12 @@ CMasternode* CMasternodeMan::GetNextMasternodeInQueueForPayment(int nBlockHeight
...
 
         //it's in the list (up to 8 entries ahead of current block to allow propagation) -- so let's skip it
-        if (masternodePayments.IsScheduled(mn, nBlockHeight)) continue;
+		// Don't check multi-tier masternodes. They can be scheduled to be paid multiple times
+        if (masternodePayments.IsScheduled(mn, nBlockHeight) && GetMasternodeTierRounds(mn.vin) == 1) continue;
 
         //it's too new, wait for a cycle
-        if (fFilterSigTime && mn.sigTime + (nMnCount * 2.6 * 60) > GetAdjustedTime()) continue;
+		// Don't check multi-tier masternodes. They are rewarded several times before being moved to the tail of the masternode queue
+        if (fFilterSigTime && mn.sigTime + (nMnCount * 2.6 * 60) > GetAdjustedTime() && GetMasternodeTierRounds(mn.vin) == 1) continue;
```
Comment (yurykk) [inline src/masternodeman.cpp]: I think this code is relevant only for a very first payment to newly installed masternode.  It should not affect masternodes that are already receiving payments.

Comment (kemi04) [reply src/masternodeman.cpp]: This function is always called with _fFilterSigTime = true_. The only time it is called with _fFilterSigTime = false_ is recursively when the network is in the process of upgrading. This means that the rest of the times time it checks if a masternode has not signed a block before all enabled masternodes could do it (_nMnCount * 2.6 *60_ - the count of enabled masternodes multiplied by 2.6 minutes).
It works with tier-1 masternodes as they go to the tail of the queue after signing blocks, but multi-tier masternodes remain in the top 10%, therefore this condition would not allow them to be rewarded for _2.6 * nMnCount_ minutes.

Comment (yurykk) [reply src/masternodeman.cpp]: SigTime represents "activation time" of the masternode, which is changing only if masternode was stopped & restarted.  In this case it is "punished" by increased time till first reward.

Therefore this part of code is relevant only for newly activated masternodes, it should not have any effect on a masternodes activated prior to  "mnCount * 2.6 * 60".

It's not relevant for tiers.

Comment (yurykk) [reply src/masternodeman.cpp]: Basically, the code is here to ensure that you will not get increased rewards by stopping masternode and starting a new one.   The time between masternode start and first reward should be significantly bigger than time between rewards for old masternodes.

That's all.

Comment (kemi04) [reply src/masternodeman.cpp]: Ok, thank you for the comments. I happened to mistake _sigTime_ with a variable storing the last time of a masternode being paid. I will review and fix the code.

**Pull request closed without merging**